compiler/plc_xml/src/xml_parser/control.rs

```diff
@@ -0,0 +1,99 @@
+use ast::ast::{AstFactory, AstNode, Operator};
+use plc_diagnostics::diagnostics::Diagnostic;
+
+use crate::model::{
+    control::{Control, ControlKind},
+    fbd::{Node, NodeIndex},
+};
+
+use super::ParseSession;
+
+impl Control {
+    pub(crate) fn transform(&self, session: &ParseSession, index: &NodeIndex) -> Result<AstNode, Diagnostic> {
+        match self.kind {
+            ControlKind::Jump => unimplemented!(),
+            ControlKind::Label => unimplemented!(),
+            ControlKind::Return => transform_return(self, session, index),
+        }
+    }
+}
+
+fn transform_return(
+    control: &Control,
+    session: &ParseSession,
+    index: &NodeIndex,
+) -> Result<AstNode, Diagnostic> {
+    let Some(ref_local_id) = control.ref_local_id else {
+        let location = session.range_factory.create_block_location(control.local_id, None);
+        return Err(Diagnostic::empty_control_statement(location));
+    };
+
+    let Some(node) = index.get(&ref_local_id) else {
+        let location = session.range_factory.create_block_location(ref_local_id, None);
+        return Err(Diagnostic::undefined_node(ref_local_id, location));
+    };
+
+    let condition = match node {
+        Node::FunctionBlockVariable(variable) => Ok(variable.transform(session)),
+        Node::Block(block) => Ok(block.transform(session, index)),
+
+        _ => {
+            let location_control = session.range_factory.create_block_location(control.local_id, None);
+            let location_other = session.range_factory.create_block_location(ref_local_id, None);
+
+            Err(Diagnostic::unexpected_nodes(location_control.span(&location_other)))
+        }
+    }?;
+
+    // XXX: Introduce trait / helper-function for negation, because we'll probably need it more often
+    let possibly_negated_condition = if control.negated {
+        let location = condition.get_location();
+        AstFactory::create_unary_expression(Operator::Not, condition, location, session.next_id())
```

Comment:
it should be  new ID IMHO, since the unary expression is a combination of the condition and an operator, the combination is a new AST element, and required a new ID

Reply:
@riederm shouldn't the create_not_expression be getting new ids?

Reply:
the operation was probably added because of line 614 here (in order to have valid annotations we need to use an existing ID here):
https://github.com/PLC-lang/rusty/blob/a28506f8a785f04297f07b4bcc9a442ad92548f8/src/codegen/generators/statement_generator.rs#L609-L624

i'd suggest we change the signature of create_not_expression to ask for an id and we fix line 614 here accordingly.